feat: add multiarch build and release

kalidax opened the pull request:
This job will

- Build and push an image to dockerhub.

- Build tofnd binary, gpg sign it, zip it, create sha256 hashes for both zip and binary, upload files to release and S3.

- For binary build the job will run on each combination of OS/arch defined in the matrix.

- For the container build it will run one time only on ubuntu-18.04/amd64

- Multiarch binary tests without docker push, upload to release and upload to s3 have been done here : https://github.com/axelarnetwork/tofnd/actions/runs/1921406385

ggutoski reviewed (approved):
Approving as codeowner but @talalashraf or @erain9 should approve before merging.

This PR subsumes #274 so we can close that after merge.

milapsheth commented on .github/workflows/build-docker-image-and-binaries.yaml:
```diff
@@ -36,59 +37,102 @@ jobs:
           submodules: recursive
 
       - name: Install Cosign
+        if: matrix.os == 'ubuntu-18.04' && matrix.arch == 'amd64'
```
Why not sign all binaries?

kalidax replied:
because it is related to docker images. Binaries are signed in gpg section

milapsheth commented on .github/workflows/build-docker-image-and-binaries.yaml:
```diff
@@ -36,59 +37,102 @@ jobs:
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+      - name: build binaries for Linux/MacOS
```
```suggestion
      - name: Build binaries
```
Perhaps?

kalidax commented:
These update will improve the build time.
- Added parallel builds : 
- 2 jobs will run in parallel : release-binaries (macos/amd64 and ubuntu/amd64) and release-docker
- I also added the check of an existing release in S3 for avoiding the overwrite

talalashraf commented:
@kalidax @milapsheth @ggutoski can we get this across the line ?
